packages/azure-http-specs/specs/azure/client-generator-core/client-initialization/client.tsp

```diff
@@ -0,0 +1,185 @@
+  name: "MultipleParamsClient",
+  service: Service,
+})
+@global.Azure.ClientGenerator.Core.clientInitialization(MultipleParamsClientOptions)
+interface MultipleParams {
+  withQuery is Service.MultipleParams.withQuery;
+  withBody is Service.MultipleParams.withBody;
+}
+
+// Scenario 3: Mix of client-level and method-level parameters
+@scenarioDoc("""
+    Client for testing a mix of client-level and method-level parameters.
+    
+    Parameters elevated to client level:
+    - name: "test-name-value" (header parameter)
+    
+    Parameters remaining at method level:
+    - region: "us-west" (query parameter)
+    
+    Expected client usage:
+    ```ts
+    const client = new MixedParamsClient({
+      name: "test-name-value"
+    });
+    
+    client.withQuery(region: "us-west");  // region stays as method param
+    client.withBody( region: "us-west", body: { name: "test-name" });  // region stays as method param
+    ```
+  """)
+@scenario
+@global.Azure.ClientGenerator.Core.client({
+  name: "MixedParamsClient",
+  service: Service,
+})
+@global.Azure.ClientGenerator.Core.clientInitialization(MixedParamsClientOptions)
+interface MixedParams {
+  withQuery is Service.MixedParams.withQuery;
+  withBody is Service.MixedParams.withBody;
+}
+
+// Scenario 4: Path parameter moved to client level
+@scenarioDoc("""
+  Client for testing a path parameter (blobName) moved to client level.
+  
+  Parameters elevated to client level:
+  - blobName: "sample-blob" (path parameter)
+  
+  Expected client usage:
+  ```ts
+  const client = new PathParamClient({
+    blobName: "sample-blob"
+  });
+  
+  // No need to pass blobName to any operations
+  client.withQuery(format: "text");  
+  client.getStandalone();
+  client.withBody("New content for sample-blob");
+  client.deleteStandalone();
+  ```
+  """)
+@scenario
+@global.Azure.ClientGenerator.Core.client({
+  name: "PathParamClient",
+  service: Service,
+})
+@global.Azure.ClientGenerator.Core.clientInitialization(PathParamClientOptions)
+interface PathParam {
+  withQuery is Service.PathParam.withQuery;
+  getStandalone is Service.PathParam.getStandalone;
+  withBody is Service.PathParam.withBody;
+  deleteStandalone is Service.PathParam.deleteStandalone;
+}
+
+// Scenario 5: Parameter aliases for better client API names
+@scenarioDoc("""
+  Client for testing the @paramAlias decorator for renaming parameters in client code.
+  
+  Parameters elevated to client level:
+  - blobName: "sample-blob" (path parameter)
+  
+  Expected client usage:
+  ```ts
+  // Elevated to client level via alias
+  client.withAliasedName();
+  
+  // Elevated to client level via original name
+  client.withOriginalName();
+  ```
+  """)
+@scenario
+@global.Azure.ClientGenerator.Core.client({
+  name: "ParamAliasClient",
+  service: Service,
+})
+interface ParamAlias {
+  withAliasedName is Service.ParamAlias.withAliasedName;
+  withOriginalName is Service.ParamAlias.withOriginalName;
+}
```

Comment:
There is no `clientInitialization` on this API?

Reply:
good catch, tahnks!